rotkehlchen/globaldb/handler.py

```diff
@@ -298,13 +298,25 @@ def get_assets_mappings(identifiers: List[str]) -> Dict[str, dict]:
         return result
 
     @staticmethod
-    def search_assets(filter_query: 'AssetsFilterQuery') -> List[Dict[str, str]]:
+    def search_assets(filter_query: 'AssetsFilterQuery', db: 'DBHandler') -> List[Dict[str, str]]:
         """Returns a list of asset details that match the search query provided."""
-        search_result = []
+        search_result: List[Dict[str, str]] = []
         query, bindings = GlobalDBHandler()._prepare_search_assets_query(filter_query)  # noqa:E501
-        with GlobalDBHandler().conn.read_ctx() as cursor:
-            cursor.execute(query, bindings)
-            for entry in cursor:
+        with db.conn.read_ctx() as cursor, GlobalDBHandler().conn.read_ctx() as global_db_cursor:
+            global_db_cursor.execute(query, bindings)
+            treat_eth2_as_eth = db.get_settings(cursor).treat_eth2_as_eth
+            found_eth = False
+            for entry in global_db_cursor:
+                if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
+                    if found_eth is False:
```

Comment:
But you didn't do what I asked exactly. Isn't this better?

```suggestion
                if found_eth is True:
                    continue

                if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
```

I mean it's better only `if treat_eth2_as_eth` which is not the default .. but it should be. We should actually change that default.